test/testUtils.ts

```diff
@@ -16,39 +16,54 @@ function getSVGParent(): D3.Selection {
   }
 }
 
-function makeFakeEvent(x: number, y: number): D3.D3Event {
-  return <D3.D3Event> <any> {
-      dx: 0,
-      dy: 0,
-      clientX: x,
-      clientY: y,
-      translate: [x, y],
-      scale: 1,
-      sourceEvent: <any> null,
-      x: x,
-      y: y,
-      keyCode: 0,
-      altKey: false
-    };
+function triggerFakeUIEvent(type: string, target: D3.Selection) {
+  var e = <UIEvent> document.createEvent("UIEvents");
+  e.initUIEvent(type, true, true, window, 1);
+  target.node().dispatchEvent(e);
 }
 
-function fakeDragSequence(anyedInteraction: any, startX: number, startY: number, endX: number, endY: number) {
-  var originalD3Mouse = d3.mouse;
-  d3.mouse = function() {
-    return [startX, startY];
-  };
-  anyedInteraction._dragstart();
-  d3.mouse = originalD3Mouse;
-  d3.event = makeFakeEvent(startX, startY);
-  anyedInteraction._drag();
-  d3.event = makeFakeEvent(endX, endY);
-  anyedInteraction._drag();
-  d3.mouse = function() {
-    return [endX, endY];
+function triggerFakeMouseEvent(type: string, target: D3.Selection, relativeX: number, relativeY: number) {
+  var clientRect = target.node().getBoundingClientRect();
+  var xPos = clientRect.left + relativeX;
+  var yPos = clientRect.top + relativeY;
+  var e = <MouseEvent> document.createEvent("MouseEvents");
+  e.initMouseEvent(type, true, true, window, 1,
+                    xPos, yPos,
+                    xPos, yPos,
+                    false, false, false, false,
+                    1, null);
```

Comment:
I think you have to pass them. THis is code that already existed, I just moved it up.